gno.land/cmd/gnoland/reset.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"os"
+	"path/filepath"
+
+	"github.com/gnolang/gno/tm2/pkg/bft/privval"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+
+	"github.com/gnolang/gno/tm2/pkg/log"
+	osm "github.com/gnolang/gno/tm2/pkg/os"
+)
+
+type resetCfg struct {
+	baseCfg
+}
+
+func (rc *resetCfg) RegisterFlags(fs *flag.FlagSet) {}
+
+// XXX: this is totally unsafe.
+// it's only suitable for testnets.
+// It could result in data loss and network disrutpion while running the node and without coordination
+func newResetAllCmd(bc baseCfg) *commands.Command {
+	cfg := resetCfg{
+		baseCfg: bc,
+	}
+
+	return commands.NewCommand(
+		commands.Metadata{
+			Name:       "unsafe-reset-all",
+			ShortUsage: "unsafe-reset-all",
+			ShortHelp:  "(unsafe) Remove all the data and WAL, reset this node's validator to genesis state",
+		},
+		&cfg,
+		func(_ context.Context, args []string) error {
+			return execResetAll(cfg, args)
+		},
+	)
+}
+
+func execResetAll(rc resetCfg, args []string) (err error) {
+	config := rc.tmConfig
+
+	return resetAll(
+		config.DBDir(),
+		config.PrivValidatorKeyFile(),
+		config.PrivValidatorStateFile(),
+		logger,
+	)
+}
+
+// resetAll removes address book files plus all data, and resets the privValdiator data.
+func resetAll(dbDir, privValKeyFile, privValStateFile string, logger log.Logger) error {
+	if err := os.RemoveAll(dbDir); err == nil {
+		logger.Info("Removed all blockchain history", "dir", dbDir)
+	} else {
+		logger.Error("Error removing all blockchain history", "dir", dbDir, "err", err)
+	}
+
+	if err := osm.EnsureDir(dbDir, 0o700); err != nil {
+		logger.Error("unable to recreate dbDir", "err", err)
+	}
+
+	// recreate the dbDir since the privVal state needs to live there
+	resetFilePV(privValKeyFile, privValStateFile, logger)
+	return nil
+}
+
+// resetState removes address book files plus all databases.
+func resetState(dbDir string, logger log.Logger) error {
+	blockdb := filepath.Join(dbDir, "blockstore.db")
+	state := filepath.Join(dbDir, "state.db")
+	wal := filepath.Join(dbDir, "cs.wal")
+	gnolang := filepath.Join(dbDir, "gnolang.db")
+
+	if osm.FileExists(blockdb) {
+		if err := os.RemoveAll(blockdb); err == nil {
+			logger.Info("Removed all blockstore.db", "dir", blockdb)
+		} else {
+			logger.Error("error removing all blockstore.db", "dir", blockdb, "err", err)
+		}
```

Comment:
could be a for loop over an array/slice of strings instead of duplicating the same code 4 times...

also I think we can just do RemoveAll and log any error while returned by that function. :shrug:

Reply:
> could be a for loop over an array/slice of strings instead of duplicating the same code 4 times...
> 
> also I think we can just do RemoveAll and log any error while returned by that function. 🤷

This part of the code is critical, we want to be explicit than abstract. Anyone looking this code knows what files we are deleting without second guessing. It increases audibility and trust of the code.  

Be explicit and less abstraction in coding style weight more in blockchain.  The end product we produce is the source code in addition to the functionality.

Reply:
Good suggestion.